edb/common/checked.py

```diff
@@ -171,6 +171,12 @@ class FrozenCheckedList(
     def __init__(self, iterable: Iterable[T] = ()) -> None:
         super().__init__()
         self._container = [self._check_type(element) for element in iterable]
+        self._hash = -1
+
+    def __hash__(self) -> int:
+        if self._hash == -1:
+            self._hash = hash(tuple(self._container))
```

Comment:
I'd also add `type(self)` to the hashed tuple.

Reply:
I mimicked what the equivalent was doing in edb.common.typed but sure, will update.